modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/engine/CommercePaymentEngineImpl.java

```diff
@@ -434,9 +434,13 @@ public CommerceOrder updateOrderPaymentStatus(
 				userId = permissionChecker.getUserId();
 			}
 
-			commerceOrder = _commerceOrderEngine.transitionCommerceOrder(
-				commerceOrder, CommerceOrderConstants.ORDER_STATUS_PENDING,
-				userId);
+			if (commerceOrder.getOrderStatus() !=
```

Comment:
can't we put this along with the other if statement?
if (paymentStatus == CommerceOrderConstants.PAYMENT_STATUS_PAID && ......)
so that we don't even need to retrieve permissionChecker ecc... if we don't need to transition the order.

Also, I know it's not you fault (really is mine), but since we are here, we should fix it with
if (permissionChecker != null)  instead or if (permissionChecker == null)

Reply:
Changes requested have been made.